release-22.1: sql: Use one BytesMonitor for all IE created by IEFactory

RichardJCai opened the pull request:
Backport
1/1 commits from #83678.
1/1 commits from #84212.

/cc @cockroachdb/release

---

Previously InternalExecutor's created via IEFactory would
create a monitor that would never be closed.

Now we have one monitor for all IEs that is closed with server
is closed.

Release note: None

RichardJCai committed:
sql: Use one BytesMonitor for all IE created by IEFactory

Previously InternalExecutor's created via IEFactory would
create a monitor that would never be closed.

Now we have one monitor for all IEs that is closed with server
is closed.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/84089)
<!-- Reviewable:end -->

RichardJCai committed:
sql: fix flakiness with InternalExecutorFactoryMemMonitor closing TestServers

We do not actually have to worry about closing the
InternalExecutorFactoryMemMonitor.

Release note: None

yuzefovich reviewed (approved):
This seems like a clean backport, right?

<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> but let's give it a few more days of baking on master.

> Reviewed 15 of 15 files at r1, 2 of 2 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/84089#-:-N6nudAk1PlKwqYn9UfM:b-lvmi3b)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @RichardJCai)



<!-- Sent from Reviewable.io -->

RichardJCai commented:
> This seems like a clean backport, right?
> 
> <img alt=":lgtm:" width="61" height="20" src="https://camo.githubusercontent.com/341bcbc4fc105a8a54332ced9ba9f1163b5fd81578777fffe7568934ec76d9d7/68747470733a2f2f72657669657761626c652e696f2f6c67746d2e706e67"> but let's give it a few more days of baking on master.
> 
> > Reviewed 15 of 15 files at r1, 2 of 2 files at r2, all commit messages.
> > _[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/84089#-:-N6nudAk1PlKwqYn9UfM:b-lvmi3b)_ status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @RichardJCai)

Yeah SGTM

RichardJCai commented:
Seems like no issues on master, merging.
